provider.go

```diff
@@ -0,0 +1,38 @@
+package summaraizer
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"text/template"
+)
+
+// AiProvider is an interface that defines an AI provider to summarize comments.
+type AiProvider interface {
```

Comment:
I suggest better name: `Summarizer` and do not use fillers like 'interface' in the comment. We can see it is an interface.
```go
// Summarizer treats reader as a stream of [Comment] and finally returns their summary.
type Summarizer interface {
}
```

Reply:
Fixed here: https://github.com/ioki-mobility/summaraizer/pull/23/commits/fa1d8bfd178f19916a49e47de6b00ba10d98cff8

Good idea!